DSESwift/DatabaseHandler.swift

```diff
@@ -114,6 +113,71 @@ class DatabaseHandler: NSObject{
         }
     }
 
+    /**
+    * Store the content of deletion request for data points in the local storage
+    * @param sensorName the name of the sensor
+    * @param sourceName the source name of the sensor
+    * @param startDate the start date to delete the data points
+    * @param endDate the end date to delete the data points
+    */
+    class func createDataDeletionRequest(sensorName: String, sourceName: String, startDate: NSDate?, endDate:  NSDate?) throws{
+        // Create data deletionrequest
+        let rlmDataDeletionRequest = DataDeletionRequest()
+        rlmDataDeletionRequest.uuid = NSUUID().UUIDString
+        rlmDataDeletionRequest.userId = KeychainWrapper.stringForKey(KEYCHAIN_USERID)!
+        rlmDataDeletionRequest.sensorName = sensorName
+        rlmDataDeletionRequest.sourceName = sourceName
+        if startDate != nil{
+            rlmDataDeletionRequest.startDate = startDate!.timeIntervalSince1970
+        }
+        if endDate != nil{
+            rlmDataDeletionRequest.endDate = endDate!.timeIntervalSince1970
+        }
+        let realm = try! Realm()
+        realm.beginWrite()
+        realm.add(rlmDataDeletionRequest, update:true)
+        
+        do {
+            try realm.commitWrite()
+        } catch {
+            throw RLMError.InsertFailed
```

Comment:
@orhan89 Swift does not allow you to call any function which can throw error without do.. catch block.